crypto/asn1/ameth_lib.c

```diff
@@ -127,9 +127,14 @@ const EVP_PKEY_ASN1_METHOD *EVP_PKEY_asn1_find_str(ENGINE **pe,
     }
     for (i = EVP_PKEY_asn1_get_count(); i-- > 0; ) {
         ameth = EVP_PKEY_asn1_get0(i);
+        if (ameth == NULL) {
```

Comment:
A quick note from a non-participating bystander: IMHO ignoring NULL pointers just cures the symptom instead of the actual problem. Having a look at the source code it seems that `EVP_PKEY_asn1_get0(i)` will never return NULL for `standard_methods`. This could only happen if there are additional  `app_methods` and even then you would have to ask yourself the question, how it can be possible that the list contains a NULL ponter.

Reply:
Ok, I overlooked that, just looked at the diffs. But nevertheless, here he definitely checks for `ameth == NULL`, not for `ameth->pem_str == NULL`.

Reply:
Ah, good point. Sirry